### PR TITLE
[CH-49] - Backend Registration & [CH-52] - Backend Logging in

### PR DESCRIPTION
**Pull request by cidkhode**
**Registration Story:** https://app.clubhouse.io/t4t/story/49/backend-registration
**Login Story:** https://app.clubhouse.io/t4t/story/52/backend-logging-in

### CODE REVIEW

File: src/main/java/com/t4t/thought4thought/entities/User.java [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
 
     private String email;
 
+    private String password;
+
+    private String passwordSalt;
+
+    private String userType;
+
+    private String interests;
```
Comment (TheMightyE) [inline src/main/java/com/t4t/thought4thought/entities/User.java]: The user may have more than one interest. Are they being stored as a string and using a delimiter to separate them?

File: src/main/java/com/t4t/thought4thought/entities/User.java [hunk 2]
```diff
@@ -1,20 +1,31 @@
...
+
+    private String fieldsOfStudy;
+
+    private String viewPoints;
```
Comment (TheMightyE) [inline src/main/java/com/t4t/thought4thought/entities/User.java]: Just as above, the user may have more than one view point.

File: src/main/java/com/t4t/thought4thought/utils/Thought4ThoughtResponseObject.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package com.t4t.thought4thought.utils;
+
+public class Thought4ThoughtResponseObject {
+    private int status; /* 0: success; 1: not successful; -1: unexpected error */
```
Comment (TheMightyE) [inline src/main/java/com/t4t/thought4thought/utils/Thought4ThoughtResponseObject.java]: *Not important*, but ENUMS could be used for the statuses. This way, if we end up adding more later, it'll be easier to keep track of them.

File: src/main/java/com/t4t/thought4thought/entities/User.java (hunk 1 above)
Comment (cidkhode) [reply src/main/java/com/t4t/thought4thought/entities/User.java]: Good catch - this is something we'll have to get back to when the mockups are enhanced. Right now it's just a simple text input field which doesn't really conform the user to any specific way of entering more than one input, as opposed to say a dropdown where we can pass the values selected through a comma separated string.

File: src/main/java/com/t4t/thought4thought/entities/User.java (hunk 2 above)
Comment (cidkhode) [reply src/main/java/com/t4t/thought4thought/entities/User.java]: Same thing as above with view points and field of study

File: src/main/java/com/t4t/thought4thought/utils/Thought4ThoughtResponseObject.java (hunk 3 above)
Comment (cidkhode) [reply src/main/java/com/t4t/thought4thought/utils/Thought4ThoughtResponseObject.java]: Makes sense. I'll add something in my next commit!

**Comment by cidkhode** [comment]
@vtran013 @Morning-Rescue @zurekoc please review when you get a chance!

**Pull request merged**